MediaBrowser.LocalMetadata/Images/LocalImageProvider.cs

```diff
@@ -106,28 +106,29 @@ private static IEnumerable<FileSystemMetadata> GetFiles(BaseItem item, bool incl
         {
             if (!item.IsFileProtocol)
             {
-                return Enumerable.Empty<FileSystemMetadata>();
+                yield break;
             }
 
             var path = item.ContainingFolderPath;
 
             // Exit if the cache dir does not exist, alternative solution is to create it, but that's a lot of empty dirs...
             if (!Directory.Exists(path))
             {
-                return Enumerable.Empty<FileSystemMetadata>();
+                yield break;
             }
 
-            if (includeDirectories)
+            var files = directoryService.GetFileSystemEntries(path).OrderBy(i => Array.IndexOf(BaseItem.SupportedImageExtensions, i.Extension ?? string.Empty));
```

Comment:
These changes are somewhat unrelated to the extras, but related to the slowness of #6906